### ISSUE TITLE
Estimating errorbars for IRTF data

### ISSUE THREAD

**Issue by fbartolic** [fbartolic/volcano#17]
@dfm  @rodluger I've tried estimating the errorbars with using the following model: sigma_i_new**2 = (alpha * sigma_i)**2 + (beta * sqrt(F_i))**2 where sigma_i is the errorbar for the i-th data point  (which I've previously estimated using a filtering procedure with the Savitzky-Golay filter) and F_i is the modeled flux.  I place a bounded normal prior N(1,0.1) for a (a>1), and a bounded normal prior on b N(0,1) (b>0). The numpyro model is implemented [here](https://github.com/fbartolic/volcano/blob/master/paper/figures/irtf_ingress_egress_fit.py#L141) and the functions for plotting the GP predictions are [here](https://github.com/fbartolic/volcano/blob/master/paper/figures/irtf_ingress_egress_plots.py#L126).  The fit of the 1998 pair of occultations is shown below:

![image](https://user-images.githubusercontent.com/18243408/105748771-7d4ef300-5f3a-11eb-93a2-9b0d5720fb01.png)

The residuals are scaled by their (inferred) errorbar.

Both alpha and beta seem to be well constrained from the data:

![image](https://user-images.githubusercontent.com/18243408/105748228-be92d300-5f39-11eb-91fe-b8eb9e82dfa8.png)

![image](https://user-images.githubusercontent.com/18243408/105748199-b470d480-5f39-11eb-9c16-122bdbeb47ca.png)
The blue histogram corresponds to the ingress light curve and the orange one to the egress light curve.

The inferred GP hyperaparamters are the following:
![image](https://user-images.githubusercontent.com/18243408/105751151-87262580-5f3d-11eb-81ee-ea720365f842.png)

It seems like the errorbars are overestimated which is why the GP predictions are flat. Any ideas on how to fix this problem? It doesn't seem like the flux dependent independent noise term is working well.

It might be a bug in the code...

**Comment by rodluger** [comment]
I haven't thought this through, but do you really need `alpha` at all? What happens if you just fix it at unity?

**Comment by fbartolic** [comment]
I tried that before, there's no difference in the results.

**Comment by rodluger** [comment]
What's the prior on the timescale and amplitude of the GP?

**Comment by dfm** [comment]
Can you also plot the posterior for the GP parameters?

It still looks to me like the issue is probably outliers, but there might be something else happening.

**Comment by fbartolic** [comment]
I figured out the problem, I forgot to rescale the GP predictions when making the plot by the same factor I used to rescale the residuals  🤦 🤦 🤦 .

Anyway, here is a fit which looks good:
![image](https://user-images.githubusercontent.com/18243408/105778804-fc592100-5f64-11eb-879e-e8c770fc8265.png)

Here I model all errorbars hierarchically assuming that each errorbar is drawn from a Half Gaussian with a same scale factor which is drawn from an Exponential distribution. So each errorbar is a free parameter. This is what the posteriors for the errorbars look like:
![image](https://user-images.githubusercontent.com/18243408/105779424-44c50e80-5f66-11eb-91d7-d3cba684c8cd.png)
![image](https://user-images.githubusercontent.com/18243408/105779443-51496700-5f66-11eb-8115-d2bd16cbf3e8.png)

I'll compare the different approaches systematically and report what I find here!
